File: solutions/java/armstrong-numbers/1/src/main/java/ArmstrongNumbers.java
import java.util.Objects;

class ArmstrongNumbers {
    boolean isArmstrongNumber(int numberToCheck) {
        char[] bytes = Objects.toString(numberToCheck).toCharArray();
        int len = bytes.length;
        int result = 0;
        for (char aByte : bytes) {
            result += (int) Math.pow(aByte - '0', len);
        }
        return result == numberToCheck;
    }
}

File: solutions/dart/armstrong-numbers/1/lib/armstrong_numbers.dart
import 'package:decimal/decimal.dart';

class ArmstrongNumbers {
  bool isArmstrongNumber(String s) {
    final power = s.length;
    var sum = Decimal.zero;

    for (final ch in s.split('')) {
      final digit = Decimal.fromInt(int.parse(ch));
      sum += _pow(digit, power);
    }

    return sum == Decimal.parse(s);
  }

  Decimal _pow(Decimal base, int exponent) {
    var result = Decimal.one;

    for (var i = 0; i < exponent; i++) {
      result *= base;
    }

    return result;
  }
}
